src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -2548,6 +2597,32 @@ private IEnumerable<string> IterateCollectionNames(Transaction tx, JsonOperation
             return result;
         }
 
+        public IEnumerable<Document> GetDocumentsByBucketFrom(DocumentsOperationContext context, int bucket, long etag)
+        {
+            var table = new Table(DocsSchema, context.Transaction.InnerTransaction);
+
+            using (GetBucketAndEtagByteString(context.Allocator, bucket, etag, out var buffer))
+            using (Slice.External(context.Allocator, buffer, buffer.Length, out var keySlice))
+            using (Slice.External(context.Allocator, buffer, buffer.Length - sizeof(long), out var prefix))
+            {
+                foreach (var result in table.SeekForwardFromPrefix(DocsSchema.DynamicKeyIndexes[DocsBucketAndEtagSlice], keySlice, prefix, 0))
+                {
+                    yield return TableValueToDocument(context, ref result.Result.Reader);
+                }
+            }
+        }
```

Comment:
I wonder if we shouldn't introduce general purpose method for getting the items by bucket. Something like `IEnumerable<SeekResult>  GetItemsByBucket(allocator,  table, dynamicindex, bucket, etag)`

Then here and in other places we could have slightly simpler code like:

```
public IEnumerable<Document> GetDocumentsByBucketFrom(DocumentsOperationContext context, int bucket, long etag)
{
    var table = new Table(DocsSchema, context.Transaction.InnerTransaction);

    foreach (var result in  GetItemsByBucket(context.Allocator, table, DocsSchema.DynamicKeyIndexes[DocsBucketAndEtagSlice], bucket, etag)
     {
            yield return TableValueToDocument(context, ref result.Result.Reader);
     }
}
```

Reply:
done